### DISCUSSION TITLE
Release schedule for v0.1.0 & v0.2.0

### DISCUSSION THREAD

**Discussion by gerhard** [dagger/dagger#1439] [General]
💡 _The current description is a conversation starter and is not final. Expect refinements and changes._

**_Last updated Jan 18, 2022 @ 4pm UTC_**

## Why are we doing this?

We are all working towards launching Dagger.

Our intent is to share the milestones with you - our amazing users 🙌🏻 - so that you know what to expect and when.

## Who is driving this?

@aluzzardi shares most of the release context, and is also the DRI - [Directly Responsible Individual](https://about.gitlab.com/handbook/people-group/directly-responsible-individuals/).

@gerhard is focusing on automating releases & understanding how they fit in the release schedule taking us to launch.

@jlongtine is converting the current changelogs into a human-friendly version & shares in [Discord #general channel](https://discord.com/channels/707636530424053791/708371226174685314).

Everyone else - including you! - are helping out by tackling issues [in this milestone](https://github.com/dagger/dagger/milestone/5), opening new ones and contributing your use-case. The more we learn about what doesn't work for you, or what you are missing, the better the launch.

The point of this GitHub discussion is to bring it all together, in a single place, for everyone to see and contribute to. 
We appreciate everyone's help! 🙌🏻

## What happened so far?

January 18, 2022 marks the beginning of dagger auto-releases. The goal is to produce publicly-available releases on a regular basis. We are starting with weekly, every Tuesday. You can find the rest of the context here https://github.com/dagger/dagger/pull/1372.

Current releases are of type "pre-release" from a [SemVer](https://semver.org/) perspective. The latest release - [v0.1.0-alpha.32](https://github.com/dagger/dagger/releases/tag/v0.1.0-alpha.32) - is a `v0.1.0` **pre-release**. Our expectation is that in the next few weeks we will release `v0.1.0`. This will mark the first concrete step towards Dagger launching.

Some of you - 👋🏻 @helderco - are already using _Europa_, the codename that we use internally for the major redesign of Dagger's core engine. https://github.com/dagger/dagger/issues/1088 is the entry-point to _Europa_ - thank you @shykes!

The latest _Europa_ changes that were shipped in [v0.1.0-alpha.32](https://github.com/dagger/dagger/releases/tag/v0.1.0-alpha.32) were announced the previous week by [@jlongtine in our Discord #general channel](https://discord.com/channels/707636530424053791/708371226174685314/931606078637080626). He also posted them on the GitHub release page.

## What happens next?

`v0.1.0-alpha.33` ships on its own & @jlongtine posts the human-friendly release notes in **Discord #general**.

We keep closing issues in [this GitHub milestone](https://github.com/dagger/dagger/milestone/5).

We figure out next steps after we answer the following:
- If the [next milestone](https://github.com/dagger/dagger/milestone/5) corresponds to `v0.1.0`, should we rename it?
- Are we going to release a `beta` or `rc` for `v0.1.0`? If so, when do we expect this to happen?
- After `v0.1.0` gets released, how will our tagging change?
  - FWIW, we are currently bumping the pre-release version, i.e. `v0.1.0-alpha.31` -> `v0.1.0-alpha.32`

## How can you help?

Let us know your thoughts on this.

[Please file an issue](https://github.com/dagger/dagger/issues/new) if you find any so that we can address as many as possible before the launch.

Have time and [want to contribute?](https://github.com/dagger/.github/blob/main/CONTRIBUTING.md) That's amazing, we look forward to it! ["Small incremental changes ftw"](https://github.com/dagger/dagger/pull/1348#issuecomment-1009628531) 🙌🏻

In case you want to get started with Dagger and don't know where, this is a great place: [Get Started with Dagger](https://docs.dagger.io/1003/get-started/) 🚀

If there is anything else that you would like to know, please ask away. Alternatively, [see you in Discord #dev](https://discord.com/channels/707636530424053791/796905486145683506) 👋🏻

---
This is follow-up to https://github.com/dagger/dagger/pull/1372#issuecomment-1014772982

**Comment by Alvise88** [comment]
I would like dagger move forward to v0.2.0 and release the latest v0.1.0 release tagged as v0.1.0; the next version can be v0.2.0-alpha.1 (removing old v0.1.0 code).
When dagger move from v0.x.0 to v0.y.0 a migration guide can be created like in [Spring Boot](https://github.com/spring-projects/spring-boot/wiki) leaving the documentation free of this burden.

When dagger v1.0.0 will be released i suggest to adhere strictly to [SemVer](https://semver.org/).
When dagger release next major version (for example v2) the community, greedily, stop supporting previous versions leaving it to a possible commercial support.

**Reply by shykes** [reply]
I agree that we should release 0.1.0 now and switch to 0.2.0alphaX in preparation for Europa (which will be 0.2.0).

**Reply by shykes** [reply]
@gerhard #1088 is an entrypoint for explaining the Europa release.

**Reply by gerhard** [reply]
> I agree that we should release 0.1.0 now

OK, let's wait for `v0.1.0-alpha.33` to be tagged automatically in the next hour - I really want to see how well this works in practice - then manually trigger tag `v0.1.0`. Who would like to join me for that?

> and switch to 0.2.0alphaX in preparation for Europa (which will be 0.2.0).

Agreed. That is a new PR which I can submit as soon as ☝🏻  happens.

**Reply by jlongtine** [reply]
@gerhard I'm sure you're aware, but it looks like the trigger release workflow happened (both automatically and via a run by you...) but didn't trigger the release workflow. It does appear that [v0.1.0](https://github.com/dagger/dagger/releases/tag/v0.1.0) was tagged by that release. I'm confused why that didn't run the release workflow. Seems like it should have. Let me know if you want/need another set of eyes to help debug this.

**Reply by gerhard** [reply]
I captured what happened in https://github.com/dagger/dagger/pull/1372#issuecomment-1015691999 after the "Trigger Release" workflow was scheduled as expected.

**Reply by aluzzardi** [reply]
Agreed.

- Last `0.1.x` release should be `0.1.0`
- Europa "alpha" releases should go out as `0.2.0-alpha.X`
- Europa eventually will go out as`0.2.0`
- `next` milestone should remain `next` for now. We'll rename it to `0.2.0-alpha.1` and close it when the time comes.
- We shouldn't attempt to name milestone against release names ahead of time because we don't know what that will be. `0.2.0-alpha.2` might happen a week after `alpha.1`, or merely hours, if we have to push a hot fix.
